Comp/src/main/java/frc/team4276/frc2024/Logging/LoggableRobotFile.java

```diff
@@ -131,4 +143,113 @@ public synchronized boolean writeToFile(String str, DebugLevel level ) {
+
+        Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (Files.isRegularFile(file)) {
+                    totalSize[0] += Files.size(file);
+                }
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+        return totalSize[0] / (1024.0 * 1024.0);
+    }
+    //finds the oldest log
+    private static String findOldestRlogFile(Path dir) throws IOException {
+        final Path[] oldestFile = {null};
+        final long[] oldestTime = {Long.MAX_VALUE};
+
+        Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (Files.isRegularFile(file) && validRlog(file.toString())) {
+                    long fileTime;
+                    try {
+                        fileTime = attrs.creationTime().toMillis();
+                    } catch (UnsupportedOperationException e) {
+                        fileTime = attrs.lastModifiedTime().toMillis();
+                    }
+
+                    if (fileTime < oldestTime[0]) {
+                        oldestTime[0] = fileTime;
+                        oldestFile[0] = file;
+                    }
+                }
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+        return oldestFile[0] != null ? oldestFile[0].toString() : null;
+    }
+
+    private static String addTimestampAndSessionID(String fileName, String sessionID) {
+        LocalTime currentTime = LocalTime.now();
+        //using colons instead of dashes causes the createfile to fail.....
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH-mm-ss");
+        String formattedTime = currentTime.format(formatter);
+
+        int dotIndex = fileName.lastIndexOf('.');
+        String namePart = dotIndex == -1 ? fileName : fileName.substring(0, dotIndex);
+        String extensionPart = dotIndex == -1 ? "" : fileName.substring(dotIndex);
+
+        return namePart + "[" + formattedTime + "][" + sessionID + "]" + extensionPart;
+    }
+    private static boolean validRlog(String fileName) {
+        return fileName.toLowerCase().endsWith(".rlog");
+    }
+    //checks if a file is being used by trying to get a lock makes sure we dont delete logs that are being used 
+    private static boolean isFileInUse(Path filePath) throws IOException {
+        try (FileChannel fileChannel = FileChannel.open(filePath, StandardOpenOption.WRITE)) {
+            fileChannel.tryLock();
+            return false; 
+        } catch (OverlappingFileLockException e) {
+            return true; 
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+    
+    private static boolean areAllFilesNotRlog(Path dir) throws IOException {
```

Comment:
Just a style thing, but I personally try to name boolean things in a positive sense, so in this case isAnyRlogFilePresent().  This way if "!" is used on it you don't have to think in double negatives